sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContextBuilder.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System;
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Interface for a builder that constructs a <see cref="DynamoDBContext"/>
+    /// </summary>
+    public interface IDynamoDBContextBuilder : IDisposable
```

Comment:
Style preference: When creating a fluent interface where each method returns back the instance of the builder the common verb names I see for methods are `Add`, `With` and `Configure`. Using `Set` implies to me the return is `void` and I don't think I can use fluent style. I would recommend for your `Set` methods use `With`.

Reply:
Understood, thx